src/isFilled.ts

```diff
@@ -224,10 +224,10 @@ export const select = isNonNullish as <Enum extends string>(
  *
  * @returns `true` if `field` is filled, `false` otherwise.
  */
-export const embed = <Data extends AnyOEmbed = AnyOEmbed & OEmbedExtra>(
-	field: EmbedField<Data>,
-): field is EmbedField<Data, "filled"> => {
-	return !!field.embed_url;
+export const embed = <Field extends EmbedField>(
+	field: Field | null | undefined,
+): field is EmbedField<Exclude<Field, Record<string, never>>, "filled"> => {
+	return isNonNullish(field) && !!field.embed_url;
```

Comment:
I had to do this weird refactor here, but feel free to tinker around, if we can find a way to revert back to the original implementation that'd be better I think. Basically:

**Original**
```typescript
export const embed = <Data extends AnyOEmbed = AnyOEmbed & OEmbedExtra>(
	field: EmbedField<Data>,
): field is EmbedField<Data, "filled"> => {
	return !!field.embed_url;
};

// Test
test("embed", (t) => {
	t.false(prismicH.isFilled.embed({})); // TypeScript: ✔️
	t.true(prismicH.isFilled.embed(prismicM.value.embed({ seed: t.title }))); // TypeScript: ✔️
});
```

**Updated**
```typescript
export const embed = <Data extends AnyOEmbed = AnyOEmbed & OEmbedExtra>(
	field: EmbedField<Data> | null | undefined,
): field is EmbedField<Data, "filled"> => {
	return isNonNullish(field) && !!field.embed_url;
};

// Test
test("embed", (t) => {
	t.false(prismicH.isFilled.embed(null)); // TypeScript: ✔️
	t.false(prismicH.isFilled.embed(undefined)); // TypeScript: ✔️
	t.false(prismicH.isFilled.embed({})); // TypeScript: ✔️
	t.true(prismicH.isFilled.embed(prismicM.value.embed({ seed: t.title }))); // TypeScript: ❌
});
```

> Somehow TypeScript failed to infer type here and you ended up with this:
>
> ```
> Argument of type 'EmptyObjectField | ((AnyOEmbed & OEmbedExtra) & { embed_url: string; html: string | null; })' is not assignable to parameter of type 'EmptyObjectField | (OEmbedBase<"photo"> & { url: string; width: number; height: number; } & OEmbedExtra & { embed_url: string; html: string | null; }) | null | undefined'.
>   Type 'OEmbedBase<"video"> & { html: string; width: number; height: number; } & OEmbedExtra & { embed_url: string; html: string | null; }' is not assignable to type 'EmptyObjectField | (OEmbedBase<"photo"> & { url: string; width: number; height: number; } & OEmbedExtra & { embed_url: string; html: string | null; }) | null | undefined'.
>     Type 'OEmbedBase<"video"> & { html: string; width: number; height: number; } & OEmbedExtra & { embed_url: string; html: string | null; }' is not assignable to type 'OEmbedBase<"photo"> & { url: string; width: number; height: number; } & OEmbedExtra & { embed_url: string; html: string | null; }'.
>       Type 'OEmbedBase<"video"> & { html: string; width: number; height: number; } & OEmbedExtra & { embed_url: string; html: string | null; }' is not assignable to type 'OEmbedBase<"photo">'.
>         Types of property 'type' are incompatible.
>           Type '"video"' is not assignable to type '"photo"'
> ```

**Workaround (PR code)**
```typescript
export const embed = <Field extends EmbedField>(
	field: Field | null | undefined,
): field is EmbedField<Exclude<Field, Record<string, never>>, "filled"> => {
	return isNonNullish(field) && !!field.embed_url;
};

// Test
test("embed", (t) => {
	t.false(prismicH.isFilled.embed(null)); // TypeScript: ✔️
	t.false(prismicH.isFilled.embed(undefined)); // TypeScript: ✔️
	t.false(prismicH.isFilled.embed({})); // TypeScript: ✔️
	t.true(prismicH.isFilled.embed(prismicM.value.embed({ seed: t.title }))); // TypeScript: ✔️
});
```